File: app/src/main/java/io/plan8/backoffice/model/api/BusinessHours.java
package io.plan8.backoffice.model.api;

import com.google.gson.annotations.SerializedName;

import io.plan8.backoffice.model.BaseModel;

/**
 * Created by chokwanghwan on 2017. 11. 28..
 */

public class BusinessHours implements BaseModel {
    @SerializedName("monday") Week monday;
    @SerializedName("tuesday") Week tuesday;
    @SerializedName("wednesday") Week wednesday;
    @SerializedName("thursday") Week thursday;
    @SerializedName("friday") Week friday;
    @SerializedName("saturday") Week saturday;
    @SerializedName("sunday") Week sunday;

    private class Week implements BaseModel{
        @SerializedName("open") boolean isOpen;
        @SerializedName("startMinuted") int startMinuted;
        @SerializedName("endMinuted") int endMinuted;
    }

    public BusinessHours() {
    }

    public Week getMonday() {
        return monday;
    }

    public Week getTuesday() {
        return tuesday;
    }

    public Week getWednesday() {
        return wednesday;
    }

    public Week getThursday() {
        return thursday;
    }

    public Week getFriday() {
        return friday;
    }

    public Week getSaturday() {
        return saturday;
    }

    public Week getSunday() {
        return sunday;
    }
}

File: app/src/main/java/io/plan8/backoffice/model/api/BusinessHours.kt
package io.plan8.backoffice.model.api

import com.google.gson.annotations.SerializedName
import io.plan8.backoffice.model.BaseModel

/**
 * Created by SSozi on 2017. 11. 20..
 */
class BusinessHours : BaseModel {
    @SerializedName("monday") lateinit var monday: Monday
    @SerializedName("tuesday") lateinit var tuesday: Tuesday
    @SerializedName("wednesday") lateinit var wednesday: WednesDay
    @SerializedName("thursday") lateinit var thursday: Thursday
    @SerializedName("friday") lateinit var friday: Friday
    @SerializedName("saturday") lateinit var saturday: Saturday
    @SerializedName("sunday") lateinit var sunday: Sunday

    class Monday : BaseModel {
        @SerializedName("open")
        var isOpen: Boolean = true
        @SerializedName("startMinuted")
        var startMinuted: Int = 0
        @SerializedName("endMinuted")
        var endMinuted: Int = 0
    }

    class Tuesday : BaseModel {
        @SerializedName("open")
        var isOpen: Boolean = true
        @SerializedName("startMinuted")
        var startMinuted: Int = 0
        @SerializedName("endMinuted")
        var endMinuted: Int = 0
    }

    class WednesDay : BaseModel {
        @SerializedName("open")
        var isOpen: Boolean = true
        @SerializedName("startMinuted")
        var startMinuted: Int = 0
        @SerializedName("endMinuted")
        var endMinuted: Int = 0
    }

    class Thursday : BaseModel {
        @SerializedName("open")
        var isOpen: Boolean = true
        @SerializedName("startMinuted")
        var startMinuted: Int = 0
        @SerializedName("endMinuted")
        var endMinuted: Int = 0
    }

    class Friday : BaseModel {
        @SerializedName("open")
        var isOpen: Boolean = true
        @SerializedName("startMinuted")
        var startMinuted: Int = 0
        @SerializedName("endMinuted")
        var endMinuted: Int = 0
    }

    class Saturday : BaseModel {
        @SerializedName("open")
        var isOpen: Boolean = true
        @SerializedName("startMinuted")
        var startMinuted: Int = 0
        @SerializedName("endMinuted")
        var endMinuted: Int = 0
    }

    class Sunday : BaseModel {
        @SerializedName("open")
        var isOpen: Boolean = true
        @SerializedName("startMinuted")
        var startMinuted: Int = 0
        @SerializedName("endMinuted")
        var endMinuted: Int = 0
    }
}